contrib/migration_script.py

```diff
@@ -0,0 +1,170 @@
+    return False
+
+
+def command_jobs(command, jobs, args):
+    data = {'command': command}
+    command_flag = True
+    for job in jobs:
+        if command == 'enable':
+            job_name = args.new_ns + '.' + job['name']
+        else:
+            job_name = args.old_ns + '.' + job['name']
+
+        if command == 'move':
+            data = {'command': command, 'old_name': args.old_ns + '.' + job['name'], 'new_name': args.new_ns + '.' + job['name']}
+            uri = urljoin(args.server, 'api/jobs')
+        else:
+            data = {'command': command}
+            uri = urljoin(args.server, 'api/jobs/' + job_name)
+
+        response = client.request(uri, data=data)
+        if response.error:
+            print(bcolors.FAIL + 'Failed to {} {}'.format(command, job_name) + bcolors.ENDC)
+            command_flag = False
+        else:
+            print(bcolors.OKGREEN + 'Succeed to {} {}'.format(command, job_name) + bcolors.ENDC)
+    return command_flag
+
+
+def ssh_command(hostname, command):
+    ssh = subprocess.Popen(["ssh", "-A", hostname, command], shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    result = ssh.stdout.readlines()
+    error = ssh.stderr.readlines()
+    if error != []:
+        print(bcolors.BOLD + 'Execute command {}: {}'.format(command, error) + bcolors.ENDC)
+    return result
+
+
+def main():
+    args = parse_args()
+    filename = args.source
+    hostname = urlparse(args.server).hostname
+    if filename.endswith(".yaml"):
+        tron_client = client.Client(args.server)
+        jobs_status = tron_client.jobs()
+
+        migration_flag = True
+        with open(filename, "r") as f:
+            jobs = yaml.load(f)['jobs']
+            for job in jobs:
+                job_name = args.old_ns + '.' + job['name']
+                migration_flag = migration_flag & check_job_if_running(jobs_status, job_name)
+
+        if migration_flag is True:
+            print(bcolors.OKBLUE + "Jobs are not running. Disable all the jobs." + bcolors.ENDC)
+        else:
+            print(bcolors.WARNING + "Some jobs are still running, abort this migration," + bcolors.ENDC)
+            return
+
+        if command_jobs('disable', jobs, args) is True:
+            print(bcolors.OKBLUE + "Jobs disabled" + bcolors.ENDC)
+        else:
+            print(bcolors.WARNING + "Some jobs aren't disabled, starting recovery" + bcolors.ENDC)
+            if command_jobs('enable', jobs, args) is True:
+                print(bcolors.OKBLUE + "Recovery successfully: Jobs are enabled again." + bcolors.ENDC)
+            else:
+                print(bcolors.WARNING + "Recovery failed: Please manually check the status of jobs" + bcolors.ENDC)
+            return
+
+        # stop cron
+        ssh_command(hostname, "sudo service cron stop")
+
+        # wait unitil yelpsoa-configs branch is merged
+        res = input("Merge and push yelpsoa-configs branch. Ready to continue? [y/n]")
```

Comment:
Is a `git mv` implied here? If so can you print out the exact git mv command to reduce mistakes?

Reply:
I was thinking "git merge --no-ff [branch]" and "git push origin master".